### PR TITLE
[CDAP-18040] Type-specific formatting for connection browser table

### PR DESCRIPTION
**Pull request by njbriggs**
Adds `DataFormatter` which is intended to be extended for other data tables.

image

### CODE REVIEW

File: app/cdap/components/Connections/Browser/GenericBrowser/BrowserTable.tsx [hunk 1]
```diff
@@ -127,14 +134,15 @@ export function BrowserTable({
...
                 </TableCell>
                 <TableCell>{entity.type}</TableCell>
-                {propertyHeaders.map((prop) => {
-                  const property = entity.properties?.[prop];
-                  const type = property?.type;
-                  const value = property?.value;
-
+                {propertyHeaders.map((header) => {
+                  const prop = entity.properties[header];
+                  console.log(prop);
```
Comment (elfenheart) [inline app/cdap/components/Connections/Browser/GenericBrowser/BrowserTable.tsx]: nit: remove

**Review by elfenheart — APPROVED** [review]

File: app/cdap/components/Connections/Browser/GenericBrowser/BrowserTable.tsx (hunk 1 above)
Comment (njbriggs) [reply app/cdap/components/Connections/Browser/GenericBrowser/BrowserTable.tsx]: Doh

**Comment by njbriggs** [comment]
Updated to remove `Intl` dependency which is not available in node 10, which we are using for unit test and build.

File: app/cdap/components/Connections/Browser/GenericBrowser/BrowserTable.tsx [hunk 2]
```diff
@@ -127,14 +134,14 @@ export function BrowserTable({
                   </div>
                 </TableCell>
                 <TableCell>{entity.type}</TableCell>
-                {propertyHeaders.map((prop) => {
-                  const property = entity.properties?.[prop];
-                  const type = property?.type;
-                  const value = property?.value;
-
+                {propertyHeaders.map((header) => {
+                  const prop = entity.properties[header];
```
Comment (elfenheart) [inline app/cdap/components/Connections/Browser/GenericBrowser/BrowserTable.tsx:138]: do Name and Type also included in properties? IIRC it's not

**Review by elfenheart — APPROVED** [review]

**Pull request merged**